sympy/physics/mechanics/kane.py

```diff
@@ -256,9 +258,10 @@ def _initialize_kindiffeq_matrices(self, kdeqs):
             k_ku = (msubs(kdeqs, qdot_zero) - f_k).jacobian(u)
             k_kqdot = (msubs(kdeqs, u_zero) - f_k).jacobian(qdot)
 
-            f_k = k_kqdot.LUsolve(f_k)
-            k_ku = k_kqdot.LUsolve(k_ku)
-            k_kqdot = eye(len(qdot))
+            if explicit_kinematics:
+                f_k = k_kqdot.LUsolve(f_k)
+                k_ku = k_kqdot.LUsolve(k_ku)
+                k_kqdot = eye(len(qdot))
```

Comment:
I'm looking at this more closely and, at the moment, I do not see what the purpose of these original lines are.

Reply:
I guess it is simply to force the block in the full mass matrix to be identity.